src/box/box.cc

```diff
@@ -1832,6 +1829,49 @@ box_set_bootstrap_leader(void)
 					  &cfg_bootstrap_leader_uuid);
 }
 
+/** Persist this instance as the bootstrap leader in _schema space. */
+static int
+box_write_bootstrap_leader(void)
+{
+	assert(instance_id != REPLICA_ID_NIL);
+	assert(!tt_uuid_is_nil(&INSTANCE_UUID));
+	return boxk(IPROTO_REPLACE, BOX_SCHEMA_ID,
+		    "[%s%s%" PRIu64 "%" PRIu32 "]", "bootstrap_leader_uuid",
+		    tt_uuid_str(&INSTANCE_UUID), fiber_time64(), instance_id);
+}
+
+int
+box_make_bootstrap_leader(void)
+{
+	if (tt_uuid_is_nil(&INSTANCE_UUID)) {
+		diag_set(ClientError, ER_UNSUPPORTED,
+			 "box.ctl.make_bootstrap_leader()",
+			 "promoting this instance before box.cfg() is called");
+		return -1;
```

Comment:
Yes, I meant the user has to issue `box.ctl.make_bootstrap_leader()` during an ongoing `box.cfg{}` call.
I tried to improve the description:
<details><summary>docbot request</summary>

Title: new bootstrap strategy - "supervised"

The `bootstrap_strategy` configuration option may now be set to
"supervised".

This strategy works as follows:
When bootstrapping a new replicaset, the nodes do not choose a bootstrap
leader automatically and instead wait for it to be appointed by the
user. The configuration will fail if no bootstrap leader is appointed
during a `replication_connect_timeout`.

In order to apppoint a bootstrap leader, the user has to issue
`box.ctl.make_bootstrap_leader()` call on the desired node. This must be
done during the initial `box.cfg()` call.

Possible ways to achieve this are:

1. In interactive mode:
```lua
fiber.create(box.cfg, desired_config)
box.ctl.make_bootstrap_leader()
```
2. Via an init script:
```lua
-- Init script:
console.listen(admin_port)
box.cfg(desired_config)
-- User console:
tarantoolctl enter admin port
> box.ctl.make_bootstrap_leader()
```

When joining a new replica with `bootstrap_strategy` = "supervised" to
an existing replica set, the replica will not choose the bootstrap
leader automatically, but will instead join to the node on which
`box.ctl.make_bootstrap_leader()` was issued last. In case such a node
isn't found after a `replication_connect_timeout`, the configuration
fails.
</details>

We can also allow the user to call `box.ctl.make_bootstrap_leader()` any time, even before the initial `box.cfg()` call. It's quite simple to implement, I just didn't think of it as an option.

The problems I see with this are:
1. Unclear what to do if user first calls `box.ctl.make_bootstrap_leader()` and then issues `box.cfg{bootstrap_strategy="auto"/"legacy"/"config"}`. Neither raising an error from box.cfg() nor ignoring the make_bootstrap_leader() call look good.
2. In general strange approach to configuration: first set some internal variable via `box.ctl.make_bootstrap_leader()` and then rely on this variable during configuration. If the user knows which node to promote **before** box.cfg, he may as well use `bootstrap_strategy "config"` and simply put self into box.cfg.bootstrap_leader.

Reply:
I don't mind any option. It simply feels like the requirement to start an admin console and do the stuff manually is an overkill.